From 75df9aebbee94a1bff98a401c572cb4361550bb0 Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Mon, 03 Jan 2022 14:38:36 +0200
Subject: [PATCH] [SW-52637] fw_upd: add versions to proto_status

Add versions to proto_status response to be read in secured way by
fw_loader.

Change-Id: I8ed6f296063d0d00f34ece1dcbcb21e9c9632af5
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/lib/preboot/module_fwupd.c b/lib/preboot/module_fwupd.c
index 033d3e4..7c49e49 100644
--- a/lib/preboot/module_fwupd.c
+++ b/lib/preboot/module_fwupd.c
@@ -27,6 +27,7 @@
 
 #define HL_FIT_LOAD_TYPE_PROP		"load-type"
 #define HL_FIT_PROTO_STATUS_PROP	"protocol-status"
+#define HL_FIT_SPI_VERSION		"spi-version"
 
 struct fwupd_state {
 	bool sending_error_flag;
@@ -55,6 +56,7 @@
 
 enum fwupd_resp_prop_type {
 	RESP_PROP_STATUS,
+	RESP_PROP_SPI_VERSION,
 	RESP_PROP_NUM
 };
 
@@ -183,6 +185,11 @@
 	resp_prop->data = &status;
 	resp_prop->size = sizeof(status);
 
+	resp_prop = &resp_props[RESP_PROP_SPI_VERSION];
+	resp_prop->name = HL_FIT_SPI_VERSION;
+	resp_prop->data = preboot_get_version(PREBOOT_BTL_VER);
+	resp_prop->size = VERSION_MAX_LEN;
+
 	err = preboot_get_mem_region(state->mregion, &mdesc);
 	if (err) {
 		preboot_puts("Failed requesting region for status FIT image\n");
